makefiles/tools/pic32prog.inc.mk

```diff
@@ -0,0 +1,43 @@
+# pic32prog flasher
+# =================
+#
+# https://github.com/sergev/pic32prog
+#
+# Allow flashing pic32 boards using:
+#  * Microchip PICkit2
+#  * Microchip PICkit3 with script firmware
+#
+#
+# PICkit-3
+# --------
+#
+# This requires changing the firmware to 'scripting mode'
+# Should be done from a Windows computer/virtual machine as described here
+#
+# https://github.com/RIOT-OS/RIOT/blob/master/dist/tools/pic32prog/doc.md
+#
+#
+# Udev rule
+# ---------
+#
+# Add yourself to the `plugdev` group, add the following `udev` rule to
+# `/etc/udev/rules.d/26-microchip.rules` and reboot.
+#
+# ```
+# # Adapted from http://en.microstickplus.com/mplabx-on-linux
+# ATTR{idVendor}=="04d8", MODE="664", GROUP="plugdev"
+# ```
+
+PIC32PROG = $(RIOTTOOLS)/pic32prog/pic32prog
+
+FLASHFILE ?= $(HEXFILE)
+
+FLASHER ?= $(PIC32PROG)
+FFLAGS  ?= $(FLASHFILE)
+
+# No reset command, but the board resets on terminal open
+RESET ?=
+RESET_FLAGS ?=
+
+# Compile pic32prog if using the one provided in RIOT
+FLASHDEPS += $(if $(findstring $(PIC32PROG),$(FLASHER)),$(PIC32PROG))
```

Comment:
Please change the FLASHDEPS logic to something like:
```
RIOT_PIC32PROG = $(RIOTTOOLS)/pic32prog/pic32prog
PIC32PROG ?= $(RIOT_PIC32PROG)
ifeq ($(PIC32PROG), $(RIOT_PIC32PROG))
  FLASHDEPS += pic32prog
endif
```

As is it doesn't work when overriding with "PIC32PROG=pic32prog make ...". This is how we usually do it with other tools (edbg, ...), not by overriding FLASHER directly.

Reply:
No need for the immediate evaluation, but indeed for the RIOT_PIC32PROG.

Reply:
(damn too fast).
But as `FLASHDEPS` will be evaluated when declaring `flash`, it cannot be defined after anyway so, does not hurt.